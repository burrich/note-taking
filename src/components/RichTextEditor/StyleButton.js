import React, { Component } from 'react';

import './styles/style-button.css';

/**
 * StyleButton component.
 */
class StyleButton extends Component {
  constructor(props) {
    super(props);
    this.style = this.props.style;
    
    // this methods binding
    this.onToggle = this.onToggle.bind(this);
  }

  onToggle(e) {
    e.preventDefault();
    this.props.onToggle(this.style.code);
  }

  render() {
    let className = 'RichEditor-styleButton';
    if (this.props.active) {
      className += ' RichEditor-activeButton';
    }

    return(
      <span className={className}
            onMouseDown={this.onToggle}>
        {this.style.label}
      </span>
    );
  }
}

export default StyleButton;
