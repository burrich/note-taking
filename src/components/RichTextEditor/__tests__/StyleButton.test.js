import React from 'react';
import { shallow } from 'enzyme';
import StyleButton from '../StyleButton';

const style = {
  code: 'BOLD',
  name: 'bold'
};
const onToggle = jest.fn();

test('Inactive style button renders correctly', () => {
  const button = shallow(
    <StyleButton 
      style={style}
      active={false}
      onToggle={onToggle} />
  );
  expect(button).toMatchSnapshot();
});

test('Active style button renders correctly', () => {
  const button = shallow(
    <StyleButton 
      style={style}
      active={true}
      onToggle={onToggle} />
  );
  expect(button).toMatchSnapshot();
});
