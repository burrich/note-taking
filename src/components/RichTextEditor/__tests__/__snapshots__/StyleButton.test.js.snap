// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active style button renders correctly 1`] = `
<span
  className="RichEditor-styleButton RichEditor-activeButton"
  onMouseDown={[Function]}
>
  label
</span>
`;

exports[`Inactive style button renders correctly 1`] = `
<span
  className="RichEditor-styleButton"
  onMouseDown={[Function]}
>
  label
</span>
`;
